src/main/scala/com.snowplowanalytics.sauna/responders/urbanairship/UAResponder.scala

```diff
@@ -0,0 +1,151 @@
+    appToListMap += (appKey -> listToIdentifierMap)
+    count += 1
+    if (count >= 10000000) {
+      urbanairship.maptoRequest(appToListMap)
+      appToListMap = Map()
+      count = 0
+    }
+  }
+
+  /**
+   * gets Option[UAInput] object from getMapEntryForLine and Maps it to the getEntry function and maps to None if gets Nothing from getMapEntryForLine
+   *
+   * @param rawline a line from the TSV file
+   * @param appToListMap the map of appKey to list map of values
+   * @return Option[l(istToIdentifierMap,appKey)] returns an Option of the tuple if it gets a UAInput object else it return None
+   */
+  def getMapEntryForLine(rawLine: String, appToListMap: Map[String, Map[String, List[Airship]]]): Option[(Map[String, List[Airship]], String)] = {
+
+    getEntryFromLine(rawLine) match {
+      case Some(uaInput) => Some(getEntry(uaInput, appToListMap))
+      case _ => None
+    }
+  }
+
+  /**
+   * makes an updated listToidentifierMap including the values from the input UAInput Object
+   *
+   * @param uaInput the UAInput object for a line in TSV
+   * @param appToListMap the map of appKey to list map of values
+   * @return (listToIdentifierMap,uaInput.appKey) the tuple of updated listToIdentfierMap including the new UAInput object and the appKey for that map
+   */
+  def getEntry(uaInput: UAInput, appToListMap: Map[String, Map[String, List[Airship]]]): (Map[String, List[Airship]], String) = {
+    val listToIdentifierMap: Map[String, List[Airship]] = if (!appToListMap.contains(uaInput.appKey)) {
+      Map[String, List[Airship]](uaInput.listName -> List(new Airship(uaInput.idType, uaInput.id)))
+    } else {
+      val listMapForApp: Map[String, List[Airship]] = appToListMap(uaInput.appKey)
+      val listMap: Map[String, List[Airship]] = if (listMapForApp.contains(uaInput.listName)) {
+        val newListEntry: Map[String, List[Airship]] = Map(uaInput.listName -> (listMapForApp(uaInput.listName) :+ new Airship(uaInput.idType, uaInput.id)))
+        listMapForApp ++ newListEntry
+      } else {
+        val newListEntry: Map[String, List[Airship]] = Map(uaInput.listName -> (List[Airship]() :+ new Airship(uaInput.idType, uaInput.id)))
+        listMapForApp ++ newListEntry
+      }
+      listMap
+    }
+    (listToIdentifierMap, uaInput.appKey)
+  }
+
+}
+
+object UAResponder {
+
+  case class Airship(identifierType: String, identifier: String)
+
+  case class UAInput(appKey: String, listName: String, idType: String, id: String)
+
+  def apply(urbanairship: urbanAirship)(implicit logger: ActorRef): Props =
+    Props(new UAResponder(urbanairship))
+
+  /**
+   * extracts the keys from a TSV line and returns it in the form of an Object
+   *
+   * @param rawLine a TSV line from the file
+   * @return Option[UAInput] returns a UAInput object if the line is in proper format with all 4 keys or else returns None
+   *
+   */
+
+  def getEntryFromLine(rawLine: String): Option[UAInput] = {
```

Comment:
Hey @rmanojcit! Well done. That is exactly what I wanted to see (although docstrings for UAInput would be nice as well).

Now let's try to get rid of mutable vars. 

But before, just to explain why I'm so obsessed with mutable vars, I want to show two examples of the effectively same algorithm in the mutable and immutable styles, so you could feel the difference. Imagine we have some collection of integers:

```scala
import scala.collection.mutable.MutableList

var collection = MutableList[Int]()

def predicate(i: Int): Boolean = ???
def transform(i: Int): String = ???
```
And we need to get some values from it (using `predicate`) and transform them (using `transform`) into new collection. How would it look like in mutable style?

```scala
def getNewList(collection: List[String]): MutableList[String] = {
  var newCollection = MutableList[String]()
  var i = 0
  while (i < collection.length) {
    if (predicate(i)) {
      newCollection + transform(i)
    }
    i = i + 1
  }
  newCollection
}
```
Do you see problems with it? We're introducing temporary, unnecessary variables like `i` and write boilerplate error-prone code just to do two simple things: `filter` and `transform`. Now lets see how it could look like in functional style.

```scala
def getNewList(collection: List[Int]): List[String] =
  collection.filter(predicate).map(transform)
```

Whoa! That's it! One single line of code to do **absolutely** the same. Which one do you prefer to read? 

Long story short: 99% of what you want to do with collections can be done using three single combinators: `map` (to transform), `filter` (well, to filter) and `fold` (to accumulate). We're going to rewrite your `AppMap` building algorithm using `fold`. Before start you can read about it [here](https://coderwall.com/p/4l73-a/scala-fold-foldleft-and-foldright) or google yourself.

Here's the plan.
1) Introduce in `UAResponder` object two [type aliases](http://twitter.github.io/effectivescala/#Types and Generics-Type aliases) `ListMap` to refer to Map of lists of `Airship` objects and `AppMap` to refer to Map of `ListMap`s
2) Write two functions: `updateListMap(listMap: ListMap, uaInput: UAInput): ListMap` and `updateAppMap(appMap: AppMap, uaInput: UAInput): AppMap`. These functions should not mutate anything, just accept two arguments and return new value, shouldn't depend on any outer values, just those you passed in, also they should be pretty short (6-10 lines of code each). Don't forget about `Option`s and patter matching, they will be helpful here.